tools/README.md

```diff
@@ -0,0 +1,73 @@
+# Readme for tools
+
+
+## Introducing grappler and tf2ngraph
+
+### Normal mode of operations
+The normal mode of operation of `ngrph_bridge` is:
+```
+import ngraph_bridge
+
+in0, in1, out0 = construct_network()
+sess = tf.Session()
+sess.run(out0, feed_dict = {in0:[2,3], in1:[4,5]})
+```
+
+The "normal" mode latches onto `Tensorflow` operations by registering `GraphOptimizationPass`s such as `NGraphRewritePass` and `NGraphVariableCapturePass`. It also register a custom op `NGraphEncapsulate`. Note that the graph rewriting and the actual execution all happens when `session.run` is called
+
+
+### Introducing `Grappler`
+`Grappler` can be thought of as a function that accepts a `graphdef` and returns a (most likely) new modified `graphdef`, although in cases of failure to transform, `Tensorflow` will silently continue to run with the input untransformed `graphdef`. Recently we have added a way to build `ngraph-bridge` with `Grappler`, by using the `--use_grappler_optimizer` flag in `build_ngtf.py`. We register the `NgraphOptimizer` as our `Grappler` pass. It pretty much does the same rewriting that `NGraphRewritePass` and `NGraphVariableCapturePass` was doing earlier, except for some subtle differences. For example, we `grappler` receives the graph in a certain stage of the `Tensorflow` execution pipeline which is different from when the `GraphOptimizationPass`s worked. Also we add `IdentityN` nodes to outputs to ensure we also capture output nodes, because by default `grappler` leaves out the `fetch`/output nodes.
```

Comment:
For example, we `grappler` ---------> For example, when `grappler`

Reply:
```suggestion
`Grappler` can be thought of as a function that accepts a `graphdef` and returns a (most likely) new modified `graphdef`, although in cases of failure to transform, `Tensorflow` will silently continue to run with the input untransformed `graphdef`. Recently we have added a way to build `ngraph-bridge` with `Grappler`, by using the `--use_grappler_optimizer` flag in `build_ngtf.py`. We register the `NgraphOptimizer` as our `Grappler` pass. It pretty much does the same rewriting that `NGraphRewritePass` and `NGraphVariableCapturePass` was doing earlier, except for some subtle differences. For example, when `grappler` receives the graph in a certain stage of the `Tensorflow` execution pipeline which is different from when the `GraphOptimizationPass`s worked. Also we add `IdentityN` nodes to outputs to ensure we also capture output nodes, because by default `grappler` leaves out the `fetch`/output nodes.
```